tools/server/server.cpp

```diff
@@ -3166,8 +3268,11 @@ struct server_context {
                         common_sampler_reset(slot.smpl);
 
                         // Process all prompt tokens through sampler system
-                        for (int i = 0; i < slot.n_prompt_tokens; ++i) {
-                            common_sampler_accept(slot.smpl, prompt_tokens[i], false);
+                        for (size_t i = 0; i < slot.cache_tokens.size(); ++i) {
+                            llama_token id = slot.prompt_tokens[i];
```

Comment:
Should this be:

```suggestion
                            llama_token id = slot.cache_tokens[i];
```

I am not sure. Maybe put a `GGML_ASSERT(slot.cache_tokens.size() == slot.prompt_tokens.size());` and if it ever triggers we can resolve it.